src/niivue.js

```diff
@@ -1326,9 +1356,204 @@ Niivue.prototype.initText = async function () {
+
+  //allocate WASM null-terminated command string
+  let cptr = this.walloc(cmd.length + 1);
+  this.linearMemory.record_malloc(cptr, cmd.length + 1);
+  let cmdstr = new Uint8Array(cmd.length + 1);
+  for (let i = 0; i < cmd.length; i++) cmdstr[i] = cmd.charCodeAt(i);
+  let cstr = new Uint8Array(this.wasmMemory.buffer, cptr, cmd.length + 1);
+  cstr.set(cmdstr);
+  console.log(cstr);
+  //allocate WASM image data
+  let nvox = nx * ny * nz * nt;
+  let ptr = this.walloc(nvox * bpv);
+  this.linearMemory.record_malloc(ptr, nvox * bpv);
+  let cimg = new Uint8Array(this.wasmMemory.buffer, ptr, nvox * bpv);
+  cimg.set(new Uint8Array(image.img.buffer));
+
+  let ok = this.processNiftiImage(
+    ptr,
+    image.hdr.datatypeCode,
+    nx,
+    ny,
+    nz,
+    nt,
+    dx,
+    dy,
+    dz,
+    dt,
+    cptr
+  );
+
+  // Unintuitive Solution: renew cimg pointer
+  // https://depth-first.com/articles/2019/10/16/compiling-c-to-webassembly-and-running-it-without-emscripten/
+  cimg = new Uint8Array(this.wasmMemory.buffer, ptr, nvox * bpv);
+
+  if (ok != 0) {
+    console.log(" -> '", cmd, " generated a fatal error: ", ok);
+    return;
+  }
+
+  let processedImage = addLayer ? image : this.volumes[imageIndex];
+
+  let isBinaryImage = cmd.toLowerCase().includes("-dog");
+  // deal with binarized image from dog
+  if (isBinaryImage) {
+    cimg = cimg.map((x) => x * 255);
+    processedImage.colorMap = "red";
+  }
+
+  switch (processedImage.hdr.datatypeCode) {
+    case processedImage.DT_UNSIGNED_CHAR:
+      processedImage.img = new Uint8Array(cimg);
+      break;
+    case processedImage.DT_SIGNED_SHORT:
+      processedImage.img = new Int16Array(cimg);
+      break;
+    case processedImage.DT_FLOAT:
+      processedImage.img = new Float32Array(cimg);
+      break;
+    case processedImage.DT_DOUBLE:
+      processedImage.img = new Float64Array(cimg);
+      break;
+    case processedImage.DT_RGB:
+      processedImage.img = new Uint8Array(cimg);
+      break;
+    case processedImage.DT_UINT16:
+      processedImage.img = new Uint16Array(cimg);
+      break;
+    case processedImage.DT_RGBA32:
+      processedImage.img = new Uint8Array(cimg);
+      break;
+    default:
+      throw "datatype " + processedImage.hdr.datatypeCode + " not supported";
+  }
+
+  //free WASM memory
+  this.linearMemory.record_free(cptr);
+  this.wfree(cptr);
+  this.linearMemory.record_free(ptr);
+  this.wfree(ptr);
+
+  // recalculate
+  processedImage.calMinMax();
+  if (addLayer) {
+    this.setVolume(processedImage, this.volumes.length);
+    this.refreshLayers(
+      processedImage,
+      this.volumes.length - 1,
+      this.volumes.length
+    );
+  } else {
+    this.refreshLayers(processedImage, imageIndex, this.volumes.length);
+  }
+  this.drawScene();
+
+  this.updateGLVolume();
+};
+
+Niivue.prototype.initWasm = async function () {
+  this.linearMemory = new LinearMemory({ initial: 2048, maximum: 2048 });
```

Comment:
If we are able to grow memory, I would change this, e.g. `initial = 256, maximum = 2048`